File: problem2.rb
#!/usr/bin/env ruby

limit = ARGV[0].to_i
puts "The limit is #{limit}"

exit unless limit


fib = Enumerator.new do |y|
	a = b = 1
	loop do
		y << a
		a, b = b, a + b
	end
end

sum = 0;

fib.each do |i|
	raise StopIteration if i > limit
	sum += i if i%2 == 0
end

puts "The result is #{sum}"

File: problem2.py
#!/usr/bin/env python

import sys

limit = int(sys.argv[1])
print 'The limit is', limit

def fibonacci():
	a = b = 1
	while True:
		yield a
		a, b = b, a + b

summation = 0
fib = fibonacci()

while True:
	fib_i = fib.next()
	if fib_i > limit:
		break
	if fib_i % 2 == 0:
		summation += fib_i

print 'The result is', summation
